tests/dags/providers/provider_api_scripts/test_provider_data_ingester.py

```diff
@@ -20,223 +20,315 @@
-        with (
-            patch.object(self.audio_store, "commit") as audio_store_mock,
-            patch.object(self.image_store, "commit") as image_store_mock,
-        ):
-            self.ingester.commit_records()
+        # process_batch is called for each successful batch
+        process_batch_mock.assert_has_calls(
+            [
+                call(EXPECTED_BATCH_DATA),
+                call(EXPECTED_BATCH_DATA),
+            ]
+        )
+        # process_batch is not called for a third time with exception
+        assert process_batch_mock.call_count == 2
+
+        # Even with the exception, records were still saved
+        assert commit_mock.called
+
+
+def test_ingest_records_uses_initial_query_params_from_dagrun_conf():
+    # Initialize the ingester with a conf
+    ingester = MockProviderDataIngester(
+        {"initial_query_params": {"has_image": 1, "page": 5}}
+    )
+
+    # Mock get_batch to halt ingestion after a single batch
+    with (
+        patch.object(ingester, "get_batch", return_value=([], False)) as get_batch_mock,
+    ):
+        ingester.ingest_records()
+
+        # get_batch is called with the query_params supplied in the conf
+        get_batch_mock.assert_called_with({"has_image": 1, "page": 5})
+
+
+def test_ingest_records_uses_query_params_list_from_dagrun_conf():
+    # Initialize the ingester with a conf
+    ingester = MockProviderDataIngester(
+        {
+            "query_params_list": [
+                {"has_image": 1, "page": 5},
+                {"has_image": 1, "page": 12},
+                {"has_image": 1, "page": 142},
+            ]
+        }
+    )
+
+    with (
+        patch.object(
+            ingester, "get_batch", return_value=(EXPECTED_BATCH_DATA, True)
+        ) as get_batch_mock,
+        patch.object(ingester, "process_batch", return_value=3),
+    ):
+        ingester.ingest_records()
+
+        # get_batch is called only three times, for each set of query_params
+        # in the list, even though `should_continue` is still True
+        assert get_batch_mock.call_count == 3
+        get_batch_mock.assert_has_calls(
+            [
+                call({"has_image": 1, "page": 5}),
+                call({"has_image": 1, "page": 12}),
+                call({"has_image": 1, "page": 142}),
+            ]
+        )
+
+
+def test_ingest_records_raises_IngestionError():
+    with (patch.object(ingester, "get_batch") as get_batch_mock,):
+        get_batch_mock.side_effect = [
+            Exception("Mock exception message"),
+            (EXPECTED_BATCH_DATA, True),  # Second batch should not be reached
+        ]
+
+        with pytest.raises(Exception) as error:
+            ingester.ingest_records()
+
+        # By default, `skip_ingestion_errors` is False and get_batch_data
+        # is no longer called after encountering an error
+        assert get_batch_mock.call_count == 1
+
+        assert str(error.value) == "Mock exception message"
```

Comment:
FWIW we can use `pytest.raises`'s `match` keyword to run this last check for us:
```suggestion
        with pytest.raises(Exception, match="Mock exception message") as error:
            ingester.ingest_records()

        # By default, `skip_ingestion_errors` is False and get_batch_data
        # is no longer called after encountering an error
        assert get_batch_mock.call_count == 1
```

Then if the exception doesn't match pytest will reraise whatever occurs in `ingest_records`.

Reply:
For some reason I always forget this! Thanks 😄